src/commands/run.ts

```diff
@@ -10,46 +10,35 @@ import {
 } from '../auth';
 import {
   addScopeToManifest,
-  isValidManifest,
+  isValidRunManifest,
 } from '../manifest';
 import { URL } from '../urls';
 import {
   ERROR,
   checkIfOnline,
   getProjectSettings,
+  getValidJSON,
   logError,
   spinner,
 } from '../utils';
 /**
  * Executes an Apps Script function. Requires clasp login --creds.
  * @param functionName {string} The function name within the Apps Script project.
  * @param cmd.nondev {boolean} If we want to run the last deployed version vs the latest code.
+ * @param cmd.params {string} JSON string of parameters to be input to function.
  * @see https://developers.google.com/apps-script/api/how-tos/execute
  * @requires `clasp login --creds` to be run beforehand.
  */
 export default async (functionName: string, cmd: { nondev: boolean; params: string }) => {
-  function IsValidJSONString(str: string) {
-    try {
-      JSON.parse(str);
-    } catch (error) {
-      throw new Error('Error: Input params not Valid JSON string. Please fix and try again');
-    }
-    return true;
-  }
 
   await checkIfOnline();
   await loadAPICredentials();
   const { scriptId } = await getProjectSettings(true);
   const devMode = !cmd.nondev; // defaults to true
   const { params: paramString = '[]' } = cmd;
-  IsValidJSONString(paramString);
-  const params = JSON.parse(paramString);
-  // Ensures the manifest is correct for running a function.
-  // The manifest must include:
-  // "executionApi": {
-  //   "access": "MYSELF"
-  // }
-  await isValidManifest();
+  const params = getValidJSON(paramString);
+
+  await isValidRunManifest();
```

Comment:
`isValidManifest` function simply tested that it was valid JSON. This makes sure it includes 
```
"executionApi": {	
    "access": "MYSELF"	
  }
```

as noted in the comment.

Reply:
Should the other access levels be allowed to? If you've deployed it to run for your entire domain, then that should be good enough for clasp run.

Reply:
Definitely. Is there documentation on the other access levels? Or should It just be a blanket of anything containing `executionApi` and `access`, regardless of value of `access`?

Reply:
The values are documented here:

https://developers.google.com/apps-script/concepts/manifests

All of the values would seem to work for the `clasp run` case, so I don't think we need to check which access is set.